oci/archive/oci_transport.go

```diff
@@ -32,9 +32,12 @@ type ociArchiveTransport struct{}
 
 // ociArchiveReference is an ImageReference for OCI Archive paths
 type ociArchiveReference struct {
-	file         string
-	resolvedFile string
-	image        string
+	file             string
+	resolvedFile     string
+	image            string
+	sourceIndex      int
+	archiveReaderRef *tempDirOCIRef
+	archiveWriterRef *tempDirOCIRef
```

Comment:
I’d expect these to be some variant of `*oci/archive.Reader` and `*oci/archive.Writer`; (Or is that `*oci/layout.{Reader,Writer}`? `*oci/internal/{Reader,Writer}`?)

Then `tempDirOCIRef` should entirely go away as a concept, in favor of the `Reader`/`Writer` types.

---

Is it this way because `tempDirOCIRef` combines _both_ an extracted directory, and a `oci/layout.ociReference`, where we want to hide the layout reference because it uses a user-invisible temporary path? I can see how continuing to use `tempDirOCIRef` can be convenient for that — but I’m not sure we need that _here_; it seems to me that `ociArchiveReference` can just hold (e.g. on the read side) a `*oci/archive.Reader`, and `newImageSource` can always combine a `*Reader.tempDirectory` + `ociArchiveReference` to create an `oci/layout.ociReference` owned only by the `ociArchiveImageSrc`, not by an `ociArchiveReference`.

Reply:
Fixed